src/mappingForPrizePool.ts

```diff
@@ -84,45 +88,60 @@ export function handleAwarded(event: Awarded): void {
     event.address.toHex(),
     _prizePool.currentPrizeId.toString()
   )
-  _prize.amount = event.params.amount
+  _prize.save()
 
+  const winner : Address = event.params.winner
+
+  const awardedControlledToken = loadOrCreateAwardedControlledToken(event.address.toHexString(), winner)
+  awardedControlledToken.amount = event.params.amount
+  
   
-  const winner = event.params.winner
-  if(winner.notEqual(Address.fromString(ZERO_ADDRESS))){
-    const existingWinners = _prize.winners || []
-    existingWinners.push(winner)
-    _prize.winners = existingWinners
+  // load ticket from prize strategy
+  let ticket : string 
+  if(_prizePool.prizeStrategy){
+    const prizeStrategy = PrizeStrategy.load(_prizePool.prizeStrategy)
+    if(prizeStrategy){
+      // multiple winners case first
+      const multipleWinnersId = prizeStrategy.multipleWinners // can be null
+      if(multipleWinnersId){
+        const multipleWinners = MultipleWinnersPrizeStrategy.load(multipleWinnersId)
+        ticket = multipleWinners.ticket
+      }
+      else{
+        const singleRandomWinnerId = prizeStrategy.singleRandomWinner 
+        const singleRandomWinner = SingleRandomWinnerPrizeStrategy.load(singleRandomWinnerId)
+        ticket = singleRandomWinner.ticket 
+      }  
+    }
   }
-  _prize.save()
-
-  // increment accumulative winnings
-  const prizePoolAccount = loadOrCreatePrizePoolAccount(event.address, event.params.winner.toHex())
-  prizePoolAccount.cumulativeWinnings = prizePoolAccount.cumulativeWinnings.plus(event.params.amount)
-  prizePoolAccount.save()
+  else{
+    ticket = null
+  }
+  awardedControlledToken.prize = _prize.id
+  awardedControlledToken.token = ticket
```

Comment:
? the token is available in the event params.  You don't need to pull in the prize strategy.